services/ingest-update-center

```diff
@@ -48,12 +66,18 @@ let ingest = {
  */
 let resolveFor = function(pluginName, graph) {
   let depends = [];
-  graph.plugins[pluginName].dependencies.forEach((dependency) => {
-    if (!dependency.optional) {
-      depends.push(dependency.name);
-      resolveFor(dependency.name, graph).forEach(d => depends.push(d));
-    }
-  });
+  let plugin = graph.plugins[pluginName];
+  let dependencies = plugin ? plugin.dependencies : undefined;
+  if (dependencies) {
+    dependencies.forEach((dependency) => {
+      if (!dependency.optional) {
+        depends.push(dependency.name);
+        resolveFor(dependency.name, graph).forEach(d => depends.push(d));
+      }
+    });
+  } else {
+    logger.info(`Skipping dependency processing for ${pluginName}, you had better make sure all deps are already in essentials.yaml`);
```

Comment:
Hack alert. As noted in Gitter, this is risky. Really the whole script needs to be rewritten: rather than load `update-center.json`, which is useless for Incrementals, it should directly download `*.hpi` and examine manifests.